solr/core/src/java/org/apache/solr/handler/component/RealTimeGetComponent.java

```diff
@@ -813,8 +813,8 @@ private static SolrInputDocument toSolrInputDocument(SolrDocument doc, IndexSche
         if ((!sf.hasDocValues() && !sf.stored()) || schema.isCopyFieldTarget(sf)) continue;
       }
       for (Object val: doc.getFieldValues(fname)) {
-        if (val instanceof Field) {
-          Field f = (Field) val;
+        if (val instanceof IndexableField) {
+          IndexableField f = materialize((IndexableField) val);
```

Comment:
you don't need this call to materialize, the next calls to `toObject` or `stringValue` will do that for you.

Reply:
Spot on!
I'm now looking to see if I can do some simplifications here; tay tuned...